airbyte-integrations/connectors/source-intercom-asaas/source_intercom/source.py

```diff
@@ -0,0 +1,532 @@
+                    {
+                        "field": self.cursor_field,
+                        "operator": ">",
+                        "value": stream_slice["start_date"],
+                    },
+                    {
+                        "field": self.cursor_field,
+                        "operator": "<",
+                        "value": stream_slice["end_date"],
+                    },
+                ],
+            },
+        }
+
+        sorting = {"sort": {"field": self.cursor_field, "order": "ascending"}}
+
+        request_body.update(query)
+        request_body.update(sorting)
+
+        pagination = {"pagination": {"per_page": 150}}
+
+        if next_page_token:
+            pagination["pagination"].update(
+                {"starting_after": next_page_token["starting_after"]},
+            )
+
+        request_body.update(pagination)
+
+        return request_body
+
+    def stream_slices(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Optional[Mapping]]:
+        """Generate slices of 30 days periods.
+
+        Args:
+            sync_mode (SyncMode): sync mode.
+            cursor_field (List[str], optional): cursor field.
+            stream_state (Mapping[str, Any], optional): stream state.
+
+        Returns:
+            Iterable[Optional[Mapping]]: stream slices
+        """
+        api_time_window_days = 30
+        extraction_end_date = pendulum.now("UTC").subtract(seconds=1)
+
+        slice_start_date = pendulum.from_timestamp(self.start_date)
+        slice_start_date = slice_start_date.subtract(seconds=1)
+
+        if stream_state:
+            slice_start_date = stream_state.get(self.cursor_field)
+
+        slices = []
+
+        while slice_start_date < extraction_end_date.subtract(seconds=1):
+            slice_end_date = slice_start_date.add(days=api_time_window_days)
+            slice_end_date = slice_end_date.add(seconds=1)
+            slice_end_date = min(slice_end_date, extraction_end_date)
+
+            slices.append(
+                {
+                    "start_date": slice_start_date.int_timestamp,
+                    "end_date": slice_end_date.int_timestamp,
+                },
+            )
+
+            slice_start_date = slice_end_date.subtract(seconds=1)
+        return slices
+
+    def next_page_token(
+        self, response: requests.Response,
+    ) -> Optional[Mapping[str, Any]]:
+        """Extract next page token from response.
+
+        Args:
+            response (requests.Response): current request response
+
+        Returns:
+            Optional[Mapping[str, Any]]: next page token
+        """
+        pages = response.json()
+        pages = pages.get("pages", {})
+        actual_page = pages['page']
+        total_pages = pages['total_pages']
+        self.logger.info("Pages {0}/{1}".format(
+            actual_page,
+            total_pages,
+        ))
```

Comment:
```suggestion
        self.logger.info(
            "Pages {0}/{1}".format(
                actual_page,
                total_pages,
            ),
        )
```
Nessas situações, colocar o argumento na próxima linha indentado.